src/js/tabs/osd.js

```diff
@@ -253,10 +253,18 @@ FONT.msp = {
 };
 
 FONT.upload = function($progress) {
-    return Promise.mapSeries(FONT.data.characters, function(data, i) {
-        $progress.val((i / FONT.data.characters.length) * 100);
-        return MSP.promise(MSPCodes.MSP_OSD_CHAR_WRITE, FONT.msp.encode(i));
-    })
+    return FONT.data.characters
+        .reduce(
+            (p, x, i) =>
+                p.then(() => {
+                    $progress.val((i / FONT.data.characters.length) * 100);
+                    return MSP.promise(
+                        MSPCodes.MSP_OSD_CHAR_WRITE,
+                        FONT.msp.encode(i),
+                    );
+                }),
+            Promise.resolve(),
```

Comment:
no argument?

Reply:
> TBH. Could probably even use for..of here with awaiting but wanted to stay closer to previious api + prefer FP more

Would prefer Functional Programming here too. Rewriting using await is more clear (to me anyways)